src/shared/remoted_op.c

```diff
@@ -79,22 +79,25 @@ void parse_uname_string (char *uname,
         // Get os_major
         if (w_regexec("^([0-9]+)\\.*", str_tmp, 2, match)) {
             match_size = match[1].rm_eo - match[1].rm_so;
-            *os_major = malloc(match_size +1 );
-            snprintf (*os_major, match_size + 1, "%.*s", match_size, str_tmp + match[1].rm_so);
+
+            if (*os_major = malloc(match_size +1 ), *os_major)
```

Comment:
I think we can use os_malloc() here. It prints an error and exit completely if there is no memory to allocate. And avoid the "if".

Reply:
DONE